src/app/clusters/door-lock-server/door-lock-server.cpp

```diff
@@ -229,8 +275,71 @@ chip::Protocols::InteractionModel::Status
 MatterDoorLockClusterServerPreAttributeChangedCallback(const chip::app::ConcreteAttributePath & attributePath,
                                                        EmberAfAttributeType attributeType, uint16_t size, uint8_t * value)
 {
-    // TODO: Implement attribute changes
-    return chip::Protocols::InteractionModel::Status::Success;
+    chip::Protocols::InteractionModel::Status res;
+
+    switch (attributePath.mAttributeId)
+    {
+    case chip::app::Clusters::DoorLock::Attributes::Language::Id: {
+        if (value[0] <= 3)
+        {
+            char lang[3 + 1] = { 0 };
+            memcpy(lang, &value[1], value[0]);
+            res = emberAfPluginDoorLockOnLanguageChange(attributePath.mEndpointId, lang);
+        }
+        else
+        {
+            res = chip::Protocols::InteractionModel::Status::InvalidValue;
+        }
+        break;
+    }
+
+    case chip::app::Clusters::DoorLock::Attributes::AutoRelockTime::Id: {
+        if (size == 4)
+        {
```

Comment:
Why must size be checked here (and not with other attributes)?

Reply:
It was asked earlier to insert sanity check here. As I understood, attribute size may change in future while Door Lock cluster will continue to use uint32_t and casting can make bad things. Other attributes are uint8_t and bool and just dereferenced instead of cast. It won't lead to invalid read.

Reply:
Ok. However, I would argue that if the size of attributes may change in the future, we should consider size checks on the other attributes as well. Even though the uint8_t attributes are a simple dereference, if their size changes in the future, the dereferenced access will yield the incorrect value on a bigendian platform.

Reply:
Good note! Fixed.